test/runtests.jl

```diff
@@ -167,6 +167,95 @@ end
     @test llkf ≈ llbf atol = 2
 end
 
+@testitem "Guided particle filter test" begin
+    using GeneralisedFilters
+    using SSMProblems
+    using StableRNGs
+    using PDMats
+    using LinearAlgebra
+    using LogExpFunctions: softmax
+    using Random: randexp
+    using Distributions
+
+    # this is a pseudo optimal proposal kernel for linear Gaussian models
+    struct LinearGaussianProposal{T<:Real} <: GeneralisedFilters.AbstractProposal
+        φ::Vector{T}
+    end
+
+    # a lot of computations done at each step
```

Comment:
Perhaps more pressing, these computations are completed twice, once for the predict step and the again for update. Not completely clear to me how to get around that though. 

We could potentially compute the proposal distribution before running the predict/update step and pass this in to each step.